### ISSUE TITLE
emmylua-ls thinks a required module could be the current file, generating false positive type-mismatch warnings

### ISSUE THREAD

**Issue by haoming-li-ling** [EmmyLuaLs/emmylua-analyzer-rust#992]
Wezterm's config file could be just name `wezterm.lua`. In that file, you are supposed to `require("wezterm")`, which is provided by Wezterm's lua runtime, and it is not referring to the current file. But then emmylua-ls thinks that the type of `require("wezterm")` is the return type of the current file, likely equating the `wezterm` module with the current file.
```lua
-- file name is `wezterm.lua`
local wezterm = require("wezterm")  ---@type Wezterm
local config = wezterm.config_builder() ---@type Config

return config
```
emmylua-ls complains that `W Cannot assign Config to Wezterm.`

**Comment by CppCXY** [comment]
Related issue: https://github.com/DrKJeff16/wezterm-types/issues/113

**Comment by CppCXY** [comment]
I suggest add `wezterm` to `---@meta` like:

<img width="578" height="101" alt="Image" src="https://github.com/user-attachments/assets/a9fd6a71-3e67-4446-945a-e15b433b8160" />

then we can get correct annotation and hint:

<img width="633" height="345" alt="Image" src="https://github.com/user-attachments/assets/aab9b72e-480d-4ffa-8298-bbe15293b938" />

**Comment by haoming-li-ling** [comment]
Which file is it that you are adding `---@meta` to? And could you explain why it works (it does work, thanks)? Also, something to consider is that LuaLS does not produce the type-mismatch warning out of the box.

**Comment by CppCXY** [comment]
Because the project contains module files with duplicate names and we perform type checking on ---@type, Lua itself doesn't provide a reliable mechanism to prevent loading a same-named module, so we must replicate that behavior ourselves.

**Comment by haoming-li-ling** [comment]
But can a file require itself as a module in Lua? I guess if there are multiple modules with the same name that are _not itself_, then what you say is certainly true.

**Comment by DrKJeff16** [comment]
Hello there, maintainer of [wezterm-types](https://github.com/DrKJeff16/wezterm-types) here.

> Which file is it that you are adding `---@meta` to? And could you explain why it works (it does work, thanks)?

@haoming-li-ling

In the case of `wezterm-types` only one file would need to be given the `---@meta wezterm` treatment ([`lua/wezterm/types/wezterm.lua`](https://github.com/DrKJeff16/wezterm-types/blob/main/lua/wezterm/types/wezterm.lua), to be precise).
I have added @CppCXY's suggestion in commit https://github.com/DrKJeff16/wezterm-types/commit/6d10943919a660038e1c9477ca37fdb8911691e3. Please test it out and tell me if that's an adequate solution for the time being.

Please excuse my lack of knowledge regarding `emmylua_ls`. I use LuaLS and make annotations for it but I don't wish to have `emmylua_ls` users excluded.
